src/relayer/RelayerClientHelper.ts

```diff
@@ -192,36 +191,35 @@ export async function constructRelayerClients(
 
 export async function updateRelayerClients(clients: RelayerClients, config: RelayerConfig): Promise<void> {
   // SpokePoolClient client requires up to date HubPoolClient and ConfigStore client.
-  const { spokePoolClients } = clients;
+  const { acrossApiClient, inventoryClient, profitClient, spokePoolClients, tokenClient } = clients;
 
   // TODO: the code below can be refined by grouping with promise.all. however you need to consider the inter
   // dependencies of the clients. some clients need to be updated before others. when doing this refactor consider
   // having a "first run" update and then a "normal" update that considers this. see previous implementation here
   // https://github.com/across-protocol/relayer-v2/pull/37/files#r883371256 as a reference.
-  await updateSpokePoolClients(spokePoolClients, [
+  const spokePoolEvents = [
     "V3FundsDeposited",
     "RequestedSpeedUpV3Deposit",
     "FilledV3Relay",
     "RelayedRootBundle",
     "ExecutedRelayerRefundRoot",
-  ]);
+  ];
+
+  // Start updates w/ no dependencies first, and wait on them last.
+  const profitClientUpdate = profitClient.update();
+  const inputTokenApprovals = config.sendingRelaysEnabled ? tokenClient.setOriginTokenApprovals() : async () => true;
```

Comment:
I think the types are a bit weird here, but this coincidentally works. Suggested a correction to match the types.

```suggestion
  const inputTokenApprovals = config.sendingRelaysEnabled ? tokenClient.setOriginTokenApprovals() : Promise.resolve();
```